Jenkinsfile

```diff
@@ -632,6 +638,72 @@ pipeline {
       }
     }
 
+    stage('Deploy Preview') {
+      when {
+        not {
+          branch 'PR-*'
+        }
+      }
+      steps {
+        setGitHubBuildStatus('nuxeo/preview', 'Deploy nuxeo Preview', 'PENDING')
+        container('maven') {
+          dir('ci/helm/preview') {
+            echo """
+            ----------------------------------------
+            Deploy Preview environment
+            ----------------------------------------"""
+            // first substitute environment variables in chart values
+            sh """
+              mv values.yaml values.yaml.tosubst
+              envsubst < values.yaml.tosubst > values.yaml
+            """
+            // second create target namespace (if doesn't exist) and copy secrets to target namespace
+            script {
+              boolean nsExist = sh(returnStatus: true, script: "kubectl get namespace ${PREVIEW_NAMESPACE}") == 0
+              String noCommentOpt = '';
+              if (nsExist) {
+                noCommentOpt = '--no-comment'
+              } else {
+                sh "kubectl create namespace ${PREVIEW_NAMESPACE}"
+              }
+              sh "kubectl create secret generic preview-docker-cfg --namespace=${PREVIEW_NAMESPACE} --from-file=.dockerconfigjson=/home/jenkins/.docker/config.json --type=kubernetes.io/dockerconfigjson --dry-run -o yaml | kubectl apply -f -"
+              // third build and deploy the chart
+              // reference branch have a different preview command. The -n name provided must be lowercase.
+              previewCommand = "jx step helm install --namespace ${PREVIEW_NAMESPACE} -n ${APP_NAME}-${PREVIEW_NAMESPACE.toLowerCase()} ."
```

Comment:
Out of curiosity, did you try to use `helm install` directly instead of `jx step helm install`?
It's worth giving it a try to rely as less as possible on the `jx` command, since we are already using a mix of pure `helm` and `jx` commands.

Reply:
Yes I tried it first for the reason you mention, not relying on jx when possible:
https://jenkins.platform.dev.nuxeo.com/blue/organizations/jenkins/nuxeo%2Fnuxeo/detail/PR-3856/52/pipeline/

Maybe it's easy to solve. I just tried jx step as a workaround as you suggested in case of and it worked and stuck with it.

Reply:
OK great, stick with `jx step helm install` then.